### PR TITLE
Isabella expenditure

### LINKED ISSUE #82: As a user, I can get the expenditure on the ingredients that I used to cook so that I can keep track of my expenditure.
Expenditure counting for ingredients

### LINKED ISSUE #83: As a user, I can get the expenditure on the ingredients that I used to cook in the recipe so that I can keep track of my expenditure.
Expenditure counting for recipes

### PR DESCRIPTION
**Pull request by cheongisabella**
My expenditure function cannot save properly have to continue working on it. But what it can currently do is:
- add to total expenditure whenever addingredients 
- add to amount used in cooking (so they'll know how much they spent just in the amount they use too cook)
- prompt to change the expenditure whenever user types deleteingredients, for eg. 1. if user delete ingredient because added accidentally, they can choose to remove the amount added to total expenditure so the program won't unnecessarily count extra, or 2. when user delete ingredient manually instead of using cookrecipe function, then prompt can offer to add amount to amountUsedInCooking variable

So since cannot save for now, everytime exit and reenter the expenditure amount restart $0. But ultimate goal is expenditure amount resets only weekly ah.

### CODE REVIEW

File: src/main/java/seedu/kitchenhelper/object/Expenditure.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+     * Instantiates singleton class.
+     *
+     * @return already created or newly created Expenditure object.
+     */
+    public static Expenditure getInstance() {
+        if (onlyInstance == null) {
+            onlyInstance = new Expenditure();
+        }
+        return onlyInstance;
+    }
```
Comment (jinfayap) [inline src/main/java/seedu/kitchenhelper/object/Expenditure.java:53]: 👍  Good work on using single singleton call for the object.

File: src/main/java/seedu/kitchenhelper/command/DeleteIngredientCommand.java [hunk 2]
```diff
@@ -135,6 +135,7 @@ public String updateNewQuantity(int newQuantity, Ingredient ingredientToDelete)
             feedbackToUser = String.format(COMMAND_FAILURE_QUANTITY, ingredientName, ingredientQuantity);
             assert ingredientQuantity != newQuantity;
         } else {
+            //Expenditure.getInstance().editExpenditure(ingredientToDelete, quantity);
```
Comment (jinfayap) [inline src/main/java/seedu/kitchenhelper/command/DeleteIngredientCommand.java:138]: Remember to remove this if not working on this implementation anymore.

**Comment by cheongisabella** [comment]
- Currently left out runtest for my parts
- Commented out edit expenditure in deleteingredientcommand due to runtest problems (because of the prompting. U guys can uncomment and see how it works if u want)
- currently ingredient price is per unit price

File: src/main/java/seedu/kitchenhelper/command/DisplayExpenditureCommand.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    public static final String COMMAND_WORD = "displayexpenditure";
+    public static final String COMMAND_DESC = "Displays the expenditure in that week.";
+    public static final String COMMAND_FORMAT = "displayexpenditure *No parameter";
+    public static final String MESSAGE_USAGE = String.format("%s: %s", COMMAND_WORD, COMMAND_DESC) + Ui.LS + String
+            .format("Example: %s", COMMAND_WORD);
+
+    public static final String EXPENDITURE = "This is the total amount you spent on buying "
+            + "Ingredients so far this week: $%.2f";
+    public static final String AMOUNT_USED_IN_COOKING = "This is the amount you actually spent "
+            + "for Ingredients used in your cooking this week: $%.2f";
```
Comment (jinfayap) [inline src/main/java/seedu/kitchenhelper/command/DisplayExpenditureCommand.java:25]: 👍 clean message

File: src/main/java/seedu/kitchenhelper/object/Expenditure.java [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        return userResponse;
+    }
+
+    public boolean isValidResponse(String userResponse) {
+        String response = userResponse;
+        if (response.equalsIgnoreCase("no") || response.equalsIgnoreCase("yes")) {
+            return true;
+        }
+        return false;
+    }
```
Comment (jinfayap) [inline src/main/java/seedu/kitchenhelper/object/Expenditure.java:165]: Do check all these functions would be used for the code later on. 
If unused, remove them in the next PR, along side with the other methods in this Expenditure class.

**Comment by jinfayap** [comment]
**Tested**
1. showing correct expenditure when adding of ingredient
2. showing correct expenditure spent when using cook recipe with ingredients with same name but different price.
3. showing correct expenditure spent when using cook recipe with ingredients. 
4. reset removed all expenditure, and reset to 0.

**Review by jinfayap — APPROVED** [review]
**Tested**
1. showing correct expenditure when adding of ingredient
2. showing correct expenditure spent when using cook recipe with ingredients with same name but different price.
3. showing correct expenditure spent when using cook recipe with ingredients.
reset removed all expenditure, and reset to 0.
4. Able to reset expenditure to 0 when reset command is used.

**Comment by yantingsanity** [comment]
Tested the following test cases that @jinfayap mentioned above and the following:
- `displayexpenditure` results were shown even after I exit and restart the application again

They are working fine! :)

File: src/main/java/seedu/kitchenhelper/command/DeleteChoreCommand.java [hunk 5]
```diff
@@ -82,22 +90,35 @@ public String deleteAll(ArrayList<Chore> choreList) {
...
-        userResponse = new Scanner(System.in).nextLine().trim();
-        while (!isValidResponse(userResponse)) {
-            System.out.println("Please enter either \"Yes\"/\"No\"");
+        try {
             userResponse = new Scanner(System.in).nextLine().trim();
+            while (!isValidResponse(userResponse)) {
+                System.out.println("Please enter either \"Yes\"/\"No\"");
+                userResponse = new Scanner(System.in).nextLine().trim();
+            }
+        } catch (NoSuchElementException e) {
```
Comment (yantingsanity) [inline src/main/java/seedu/kitchenhelper/command/DeleteChoreCommand.java:107]: Good catch of exception!

File: src/main/java/seedu/kitchenhelper/notification/ChoreNotification.java [hunk 6]
```diff
@@ -7,12 +7,23 @@
...
+ */
 public class ChoreNotification {
 
     public static final String DEADLINE_NOTIFICATION = "The deadline for these chores are approaching!\n";
     public static final String EMPTY_NOTIFICATION = "You have no deadlines upcoming in the next 3 days.\n";
     public static final String OVERDUE_NOTIFICATION = "The deadline for these chores are overdue!\n";
 
+    /**
+     * For those chores containing deadline specified by date,
+     * check for expired deadlines or deadlines upcoming in 3 days,
```
Comment (yantingsanity) [inline src/main/java/seedu/kitchenhelper/notification/ChoreNotification.java:21]: Addition of JavaDoc made me understand the methods much more clearer! Good job!

**Review by yantingsanity — COMMENTED** [review]
LGTM!

**Pull request merged**